config.js

```diff
@@ -1,17 +1,19 @@
 require('dotenv').config()
 
-const production = process.env.NODE_ENV === 'production'
+const generateGet = (env) => (name, fallback, { requireInProduction, noFallbackInProduction } = {}) => {
+  if (env === 'test') return fallback
 
-const get = (name, fallback, { requireInProduction, noFallbackInProduction } = {}) => {
   if (process.env[name]) return process.env[name]
 
-  if (production && noFallbackInProduction) return undefined
+  if (env === 'production' && noFallbackInProduction) return undefined
 
-  if (fallback !== undefined && !(production && requireInProduction)) return fallback
+  if (fallback !== undefined && !(env === 'production' && requireInProduction)) return fallback
 
   throw new Error(`Missing env var ${name}`)
 }
 
+const get = generateGet(process.env.NODE_ENV)
```

Comment:
The example here flips the concept of returning functions I was aiming for. In mine I was going for partial application as I am used to working with curried functions. This allows you to pass one of the arguments to the function, and be returned a another function that is expecting the rest of the arguments. For me that's simpler to read than a function that does some stuff, then returns another function, but I think thats just a stylistic preference.

I was passing the process.env as a partially applied argument to allow for functional scope to avoid the accessing of module scope of `process.env`. This is useful if we want to test `generateGet` later down the line as it is now pure and does not rely on module scope.

Let me know if this reasoning makes sense, I was also going for a minimal line change. Thinking about this file a little deeper, I think my preference would be for our config to be more explicit, rathar than hidden behind obscure functions. To do this we could have something like:

```
defaultConfig = {
}

productionConfig = {
  apiUrl: "https://real-api"
}

devConfig = {
  apiUrl: "localhost:3000"
}

getConfig = (env) => {
  if (env === "test") return null
  if (env === "production") return {...defaultConfig, ...productionConfig}
  if (env === "development") return {...defaultConfig, ...devConfig}
}

getConfig(process.env)
```

LMK what you think :)